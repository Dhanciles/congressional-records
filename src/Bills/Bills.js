import React from 'react'; 
import '../styles/Bills.scss'; 
import { connect } from 'react-redux';

export const Bills = () => {
  return (
    <article className="bill-container">
      <p className="bill-id">bill-id</p>
      <p className="chamber">Chamber - House</p>
      <p className="sponsor">Sponsor</p>
      <img src='./icons/bookmark.svg' alt='bookmark'/>
      <h4 className="bill-title">Title</h4>
      <p className="commitees">Commitees</p>
      <p className="status"> Active - False </p>
      <p className="vote-infromation">Vote Information</p>
    </article>
  )
}
