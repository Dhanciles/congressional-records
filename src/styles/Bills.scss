@import "main";

* {
  box-sizing: border-box; 
  margin: 0; 
  padding: 0; 
}

.bill-container {
  display: flex;
  flex-direction: column; 
  background-color: $color-grey; 
  height: 18vh; 
  width: 65%; 
  margin: 20px 0 20px 0;  
  border-radius: 15px; 
  padding: 10px; 
  p {
    font-family: $font-family-b; 
    font-weight: 600; 
    font-size: 2rem; 
    color: #000000; 
  }
  .bookmark-icon {
    height: 30px; 
    width: 30px; 
  }
  .top-level-card {
    display: flex; 
    justify-content: space-between; 
    margin: 20px 
  }
  .bill-title-container {
    text-align: center; 
  }
  .bottom-level-card {
    display: flex; 
    justify-content: space-between;
    text-align: center; 
  }
}

