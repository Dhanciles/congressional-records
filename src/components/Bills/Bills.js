import React, { Component } from 'react'; 
import '../../styles/Bills.scss'; 
import PropTypes from 'prop-types'

export class Bills extends Component {
  constructor() {
    super()
  }
  render() {
    const { billId, sponsor, title, lastVote, committee } =  this.props
    const status = lastVote === null ? false : true
    return (
      <article className="bill-container">
        <div className="top-level-card">
          <p className="bill-id">{billId}</p>
          <p className="chamber">Chamber - House</p>
          <p className="sponsor">{sponsor}</p>
          <img  className="bookmark-icon"src='./icons/bookmark.svg' alt='bookmark'/>
        </div>
        <div className="bill-title-container">
          <h4 className="bill-title">{title}</h4>
        </div>
        <div className="bottom-level-card">
          <p className="commitees">{committee}</p>
          <p className="status"> Active - {status} </p>
          <p className="vote-infromation">{lastVote}</p>
        </div>
      </article>
    )
  }
}

Bills.propTypes = {
  billId: PropTypes.string, 
  sponsor: PropTypes.string, 
  title: PropTypes.string, 
  committee: PropTypes.string, 
  active: PropTypes.string, 
  lastVote: PropTypes.string
}



