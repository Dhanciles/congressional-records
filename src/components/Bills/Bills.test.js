import React from 'react'; 
import { shallow } from 'enzyme'; 
import { Bills }  from './Bills'

describe('Bills', () => {
  it('should match the snapshot', () => {
    let mockBillId = "hconres283-110"
    let mockSponsor = "Rep.Donald M. Payne, NJ"
    let mockTitle = "Calling for a peaceful resolution to the current electoral crisis in Kenya."
    let mockCommitee = "House Foreign Affairs; Senate Foreign Relations"
    let mockActive = null
    let mockLastVote = "2008-02-07"

    const wrapper = shallow(<Bills
                              billId={mockBillId}
                              sponsor={mockSponsor}
                              title={mockTitle}
                              committee={mockCommitee}
                              active={mockActive}
                              lastVote={mockLastVote}
                            />)

    expect(wrapper).toMatchSnapshot()
  })
})