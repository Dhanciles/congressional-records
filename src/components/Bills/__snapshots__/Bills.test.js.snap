// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bills should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bills
    active={null}
    billId="hconres283-110"
    committee="House Foreign Affairs; Senate Foreign Relations"
    lastVote="2008-02-07"
    sponsor="Rep.Donald M. Payne, NJ"
    title="Calling for a peaceful resolution to the current electoral crisis in Kenya."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="bill-id"
        >
          hconres283-110
        </p>,
        <p
          className="chamber"
        >
          Chamber - House
        </p>,
        <p
          className="sponsor"
        >
          Rep.Donald M. Payne, NJ
        </p>,
        <img
          alt="bookmark"
          src="./icons/bookmark.svg"
        />,
        <h4
          className="bill-title"
        >
          Calling for a peaceful resolution to the current electoral crisis in Kenya.
        </h4>,
        <p
          className="commitees"
        >
          House Foreign Affairs; Senate Foreign Relations
        </p>,
        <p
          className="status"
        >
           Active - 
          true
           
        </p>,
        <p
          className="vote-infromation"
        >
          2008-02-07
        </p>,
      ],
      "className": "bill-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hconres283-110",
          "className": "bill-id",
        },
        "ref": null,
        "rendered": "hconres283-110",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Chamber - House",
          "className": "chamber",
        },
        "ref": null,
        "rendered": "Chamber - House",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rep.Donald M. Payne, NJ",
          "className": "sponsor",
        },
        "ref": null,
        "rendered": "Rep.Donald M. Payne, NJ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "bookmark",
          "src": "./icons/bookmark.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Calling for a peaceful resolution to the current electoral crisis in Kenya.",
          "className": "bill-title",
        },
        "ref": null,
        "rendered": "Calling for a peaceful resolution to the current electoral crisis in Kenya.",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Foreign Affairs; Senate Foreign Relations",
          "className": "commitees",
        },
        "ref": null,
        "rendered": "House Foreign Affairs; Senate Foreign Relations",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Active - ",
            true,
            " ",
          ],
          "className": "status",
        },
        "ref": null,
        "rendered": Array [
          " Active - ",
          true,
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2008-02-07",
          "className": "vote-infromation",
        },
        "ref": null,
        "rendered": "2008-02-07",
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="bill-id"
          >
            hconres283-110
          </p>,
          <p
            className="chamber"
          >
            Chamber - House
          </p>,
          <p
            className="sponsor"
          >
            Rep.Donald M. Payne, NJ
          </p>,
          <img
            alt="bookmark"
            src="./icons/bookmark.svg"
          />,
          <h4
            className="bill-title"
          >
            Calling for a peaceful resolution to the current electoral crisis in Kenya.
          </h4>,
          <p
            className="commitees"
          >
            House Foreign Affairs; Senate Foreign Relations
          </p>,
          <p
            className="status"
          >
             Active - 
            true
             
          </p>,
          <p
            className="vote-infromation"
          >
            2008-02-07
          </p>,
        ],
        "className": "bill-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hconres283-110",
            "className": "bill-id",
          },
          "ref": null,
          "rendered": "hconres283-110",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Chamber - House",
            "className": "chamber",
          },
          "ref": null,
          "rendered": "Chamber - House",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rep.Donald M. Payne, NJ",
            "className": "sponsor",
          },
          "ref": null,
          "rendered": "Rep.Donald M. Payne, NJ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "bookmark",
            "src": "./icons/bookmark.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Calling for a peaceful resolution to the current electoral crisis in Kenya.",
            "className": "bill-title",
          },
          "ref": null,
          "rendered": "Calling for a peaceful resolution to the current electoral crisis in Kenya.",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Foreign Affairs; Senate Foreign Relations",
            "className": "commitees",
          },
          "ref": null,
          "rendered": "House Foreign Affairs; Senate Foreign Relations",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Active - ",
              true,
              " ",
            ],
            "className": "status",
          },
          "ref": null,
          "rendered": Array [
            " Active - ",
            true,
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2008-02-07",
            "className": "vote-infromation",
          },
          "ref": null,
          "rendered": "2008-02-07",
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
